### ISSUE TITLE
Turn notification text and TTS isn't localized

### ISSUE THREAD

**Issue by zyphlar** [organicmaps/organicmaps#8607]
The map itself will show street names in the selected locale however a user reports that on iOS with Marathi as an app and TTS language (there is no iOS Marathi system level option) street names are shown in the turn notification and spoken over TTS using the value of the name tag instead of the name:mr tag.

Reproduce by initiating a turn on iOS with Marathi selected onto a road that has an English name but Marathi name:mr

**Comment by biodranik** [comment]
It should be reproducible with any other language too. I don't remember that TTS-selected language was used to retrieve street names.

**Comment by fabwu** [comment]
@zyphlar Is this what you selected?

<img src="https://github.com/user-attachments/assets/651c674d-4753-4a74-b004-11695cf88f5b" width="25%">

If I select this language (sorry I can't read this amazing alphabet), the turn notification language is set to `hi` (Hindi?) and not `mr`. 

Could that be the problem?

**Comment by zyphlar** [comment]
@fabwu I'm not sure which issue you're encountering, `ไทย` is Thai (`th`)
This issue is about feeding localized street names as a priority into the turn notification system, just like they're displayed with language priority on the map. Currently it seems like names are displayed and announced based on their `name` value or similar, which is often not localized (for example, `name` in OSM India is always English by custom despite India having many mutually exclusive languages.)

**Comment by fabwu** [comment]
Ah ok then I was on the wrong track. Forget what I wrote before :)

**Comment by parnikkapore** [comment]
Speaking of `th`, navigating in Thailand with the TTS language set to English would lead to street names not being read properly. (Most TTS engines skip all Thai characters, i.e. anything but the numbers; Google, uh, tries its best.)

**Comment by zyphlar** [comment]
@parnikkapore good point. Do you know, does Google's English pronunciation of Thai words sound like a UK/US tourist? Because if so, that could actually be a feature not a bug. (I "know Japanese" but not really, I need things spelled out at a below-kindergarten level. Using a familiar accent can help with that.)

**Comment by parnikkapore** [comment]
Google Maps does that; Google TTS is often completely unintelligible (maybe attempting to read a *transliteration* as an English sentence, which doesn't work? I'm curious myself)

**Comment by biodranik** [comment]
Another locale-specific case: a user expecting Spanish navigation reports that turn and roundabout destinations around 43.535809, -5.831697 (AS-110/AS-118 toward Luanco) are spoken in Asturian/Bable instead of Spanish. This is a useful test for applying the selected map/TTS language to names used in turn notifications.
